pkg/Microsoft.Windows.Compatibility/tests/Microsoft.Windows.Compatibility.Validation.csproj

```diff
@@ -1,19 +1,59 @@
 <Project Sdk="Microsoft.NET.Sdk">
+
+  <Import Project="../../../eng/Versions.props" />
+
   <ItemGroup>
     <PackageReference Include="Microsoft.Windows.Compatibility" Version="$(CompatibilityPackageVersion)" />
-   </ItemGroup>
-   <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp3.0'">
-    <PackageReference Include="Microsoft.Private.CoreFx.NETCoreApp" Version="$(PrivateCorefxPackageVersion)" />
-    <PackageReference Include="Microsoft.AspNetCore.App" Version="3.0.0-*" />
-    <PackageReference Include="Microsoft.DesktopUI.App" Version="3.0.0-*" />
-    <PackageReference Include="System.Windows.Extensions" Version="4.6.0-*" />
-   </ItemGroup>
-   <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp2.1'">
+  </ItemGroup>
+  <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp3.0'">
+    <PackageReference Include="Microsoft.Private.CoreFx.NETCoreApp" Version="$(PrivateCorefxNetCoreAppPackageVersion)" />
+    <PackageReference Include="Microsoft.AspNetCore.App.Ref" Version="$(MicrosoftAspNetCoreAppRefPackageVersion)" ExcludeAssets="All" GeneratePathProperty="true" />
+    <Reference Include="$(PkgMicrosoft_AspNetCore_App_Ref)\ref\$(TargetFramework)\*.dll" />
+    <PackageReference Include="Microsoft.WindowsDesktop.App.Ref" Version="$(MicrosoftWindowsDesktopAppRefPackageVersion)" ExcludeAssets="All" GeneratePathProperty="true"  />
+    <Reference Include="$(PkgMicrosoft_WindowsDesktop_App_Ref)\ref\$(TargetFramework)\*.dll" />
+    <PackageReference Include="System.Windows.Extensions" Version="$(SystemWindowsExtensionsVersion)" />
+  </ItemGroup>
+  <ItemGroup Condition="'$(TargetFramework)' == 'uap10.0.16300'">
+    <PackageReference Include="Microsoft.Private.CoreFx.UAP" Version="$(PrivateCorefxUAPPackageVersion)" />
+  </ItemGroup>
+  <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp2.1'">
     <PackageReference Include="Microsoft.NETCore.App" Version="2.1.5" />
     <PackageReference Include="Microsoft.AspNetCore.All" Version="2.1.5" />
-   </ItemGroup>
-   <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp2.0'">
+  </ItemGroup>
+  <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp2.0'">
     <PackageReference Include="Microsoft.AspNetCore.All" Version="2.0.9" />
     <PackageReference Include="Microsoft.NETCore.App" Version="2.0.9" />
   </ItemGroup>
+
+  <PropertyGroup>
+    <TargetFrameworks>uap10.0.16300;netcoreapp3.0</TargetFrameworks>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="$(TargetFramework.StartsWith('uap'))">
```

Comment:
You may be able to remove this now that you aren't relying on publish, that said it doesn't hurt anything.  If you keep it, add a comment indicating that it's teaching the SDK how to parse UAP* targetframework.

Reply:
actually no, this is still needed since we do run ResolveReferences.